src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectStrategyService.java

```diff
@@ -34,18 +41,31 @@
     private final JenkinsIntLogger logger;
     private final String remoteTempWorkspacePath;
     private final JenkinsProxyHelper jenkinsProxyHelper;
+    private final JenkinsConfigService jenkinsConfigService;
 
-    public DetectStrategyService(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsProxyHelper, String remoteTempWorkspacePath) {
+    public DetectStrategyService(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsProxyHelper, String remoteTempWorkspacePath, JenkinsConfigService jenkinsConfigService) {
         this.logger = logger;
         this.jenkinsProxyHelper = jenkinsProxyHelper;
         this.remoteTempWorkspacePath = remoteTempWorkspacePath;
+        this.jenkinsConfigService = jenkinsConfigService;
     }
 
-    public DetectExecutionStrategy getExecutionStrategy(IntEnvironmentVariables intEnvironmentVariables, OperatingSystemType operatingSystemType, String remoteJdkHome) {
-        DetectExecutionStrategy detectExecutionStrategy;
+    public DetectExecutionStrategy getExecutionStrategy(IntEnvironmentVariables intEnvironmentVariables, OperatingSystemType operatingSystemType, String remoteJdkHome, DetectDownloadStrategy detectDownloadStrategy)
+        throws IntegrationException {
+        if (detectDownloadStrategy == null || detectDownloadStrategy instanceof InheritFromGlobalDownloadStrategy) {
+            DetectGlobalConfig detectGlobalConfig = jenkinsConfigService.getGlobalConfiguration(DetectGlobalConfig.class)
+                                                        .orElseThrow(() -> new DetectJenkinsException("Unable to identify Detect Global Configuration."));
```

Comment:
If we're not able to find the Detect Global Configuration, that may mean it hasn't been configured. We should tell users in terms they understand that we couldn't find the Synopsys Detect configuration in the system configuration, and that they should check their system configuration.